plugins/woocommerce-blocks/assets/js/base/components/cart-checkout/shipping-rates-control/index.tsx

```diff
@@ -70,14 +70,37 @@ const ShippingRatesControl = ( {
 	renderOption,
 	context,
 }: ShippingRatesControlProps ): JSX.Element => {
+	const shippingRatesRateCount = useRef(
+		getShippingRatesRateCount( shippingRates )
+	);
+	const shippingRatesPackageCount = useRef(
+		getShippingRatesRateCount( shippingRates )
+	);
+
 	useEffect( () => {
 		if ( isLoadingRates ) {
 			return;
 		}
+
+		const newShippingRatesRateCount =
+			getShippingRatesRateCount( shippingRates );
+		const newShippingRatesPackageCount =
+			getShippingRatesPackageCount( shippingRates );
+
+		if (
+			shippingRatesRateCount.current === newShippingRatesRateCount &&
+			shippingRatesPackageCount.current === newShippingRatesPackageCount
+		) {
+			return;
+		}
```

Comment:
Is this code to prevent the same values being spoken multiple times? I think what you have is fine but it would be my preference to avoid the refs and instead use the `usePrevious` hook.

```
import { usePrevious } from '@woocommerce/base-hooks';
```

This will let you store the previous value to a variable, which you can then compare in the `useEffect` hook to ensure it only speaks when a change has occurred. What do you think?

Reply:
@mikejolley I didn't know about the `usePrevious` hook. I think using it makes sense to keep consistency with the rest of the codebase.

I have made the changes, please review it:

- [Use usePrevious instead of useRef to track changing in values](https://github.com/woocommerce/woocommerce/pull/53494/commits/9f7516e2518331aeaa1b67fd875c595960844ed2)

Thanks!

cc: @dkotter